File: Ruby/lib/bubble_sort.rb
require 'pry-state'

class Bubble

  def sort(arr)
    index_count = arr.length
    loop do
      swapped = false
      (index_count-1).times do |i|
        if arr[i] > arr[i+1]
          arr[i], arr[i+1] = arr[i+1], arr[i]
          swapped = true
        end
      end
      break if not swapped
    end
    arr
  end

end

File: Python/bubble_sort/bubble_sort.py
from IPython import embed

class Bubble:

  def sort(self, arr):
    for num in range(len(arr)-1, 0 ,-1):
      for n in range(num):
        if arr[n] > arr[n+1]:
          arr[n] , arr[n+1] = arr[n+1], arr[n]
    return arr